### PR TITLE
Allow overridden and new methods on hubs

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/SignalR/issues/333

@anurse @moozzyk @mikaelm12

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs [hunk 1]
```diff
@@ -309,23 +309,65 @@ private void DiscoverHubMethods()
         {
             var typeInfo = typeof(THub).GetTypeInfo();
 
-            foreach (var methodInfo in typeInfo.DeclaredMethods.Where(m => IsHubMethod(m)))
+            // used for checking against methods in base classes
+            var methods = new Dictionary<string, MethodInfo>();
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs]: This doesn't look right at all. Did you look at MVC by any chance?

Comment (anurse) [inline src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs]: My main question would be if MVC blocks the use of the `new` keyword intentionally or if it is an unintentional omission. It does seem odd to me that we would half-support derived classes by supporting overriding but not shadowing.

**Comment by BrennanConroy** [comment]
:up: :date: No more `new` methods

**Comment by BrennanConroy** [comment]
@rynowak Could you take a look?

**Pull request merged**